server/service/osquery_utils/queries_test.go

```diff
@@ -441,3 +442,17 @@ func TestDirectIngestChromeProfiles(t *testing.T) {
 	require.NoError(t, err)
 	require.True(t, ds.ReplaceHostDeviceMappingFuncInvoked)
 }
+
+func TestDangerousReplaceQuery(t *testing.T) {
+	queries := GetDetailQueries(&fleet.AppConfig{HostSettings: fleet.HostSettings{EnableHostUsers: true}}, config.FleetConfig{})
+	originalQuery := queries["users"].Query
+
+	err := os.Setenv("FLEET_DANGEROUS_REPLACE_users", "select * from blah")
```

Comment:
Nit: `t.Cleanup(os.Clearenv)` (to not leave variable set in case some of the checks below fail)

Reply:
Also, instead of `os.Clearenv` consider unsetting the set variable only (to not mess up with other env variables set on the test run)